### PR TITLE
Change default value for copy_all_values in Potential.calculate() to …

### PR DESCRIPTION
**Pull request by jameskermode**
…False, and add GAP local variance

### CODE REVIEW

**Comment by gabor1** [comment]
do you want to rebase this so that the Travis can check it ?

**Comment by jameskermode** [comment]
sure - how do I do that, `git rebase public` from the branch?

**Comment by gabor1** [comment]
uh... dunno, I usually get git-help from you! you want your branch to get the later commits from public. maybe look at how the previous pull request did it two days ago after the travis was fixed?

**Comment by jameskermode** [comment]
should be done now, let's see.

File: quippy/quippy/potential.py [hunk 1]
```diff
@@ -283,7 +283,7 @@ def calculate(self, atoms=None, properties=None, system_changes=None,
                 _v_atom = self.atoms.get_volume() / self._quip_atoms.n
             self.results['stresses'] = -np.copy(_quip_properties['local_virial']).T.reshape((self._quip_atoms.n, 3, 3),
                                                                                             order='F') / _v_atom
-        if isinstance(copy_all_results, bool) and copy_all_results:
+        if isinstance(copy_additional_results, bool) and copy_additional_results:
```
Comment (bernstei) [inline quippy/quippy/potential.py]: Does it make sense to do this only when `copy_additional_results` is `bool` and `True`, but not fail if it's not `bool`?  If the only meaningful value is bool true, I'd think we want to fail if it's not bool.  I think failing for other values that would normally make the `if` clause occur would be very surprising behavior (unless this is a common python or ASE behavior that I'm just not aware of).

**Review by bernstei — COMMENTED** [review]
one question

File: quippy/quippy/potential.py (hunk 1 above)
Comment (jameskermode) [reply quippy/quippy/potential.py]: I agree a simple `if copy_additional_results:` would be more idiomatic. @stenczelt, the type check was from your previous version of this code, do you remember why its there?

Comment (stenczelt) [reply quippy/quippy/potential.py]: I agree, simplify it

**Comment by jameskermode** [comment]
ok to merge this now? @noambernstein @stenczelt

File: quippy/quippy/potential.py [hunk 2]
```diff
@@ -194,7 +194,6 @@ def calculate(self, atoms=None, properties=None, system_changes=None,
         val = _check_arg(local_virial)
         if val == 'y':
             properties += ['local_virial']
-        elif val == 'add':
             properties += ['local_virial']
```
Comment (stenczelt) [inline quippy/quippy/potential.py]: this line is duplicated here

Comment (jameskermode) [reply quippy/quippy/potential.py]: good spot. Not sure why I changed that line. I'll remove it from the PR.

**Review by stenczelt — CHANGES_REQUESTED** [review]
I think this is all reasonable, apart from that one line change I agree with all of it.

**Comment by stenczelt** [comment]
All good!

**Review by stenczelt — APPROVED** [review]

**Comment by bernstei** [comment]
Works for me now.  I'm happy to merge.

**Comment by jameskermode** [comment]
Ok, great. I’ll merge this now then.

**Pull request merged**